### ISSUE TITLE
Improving Travis builds and notifications

### ISSUE THREAD

**Issue by AlexisDrogoul** [gama-platform/gama#1876]
### Steps to reproduce
1. Observe a build made by Travis (https://travis-ci.org/gama-platform/gama)

2. When it is successful and finished, some time passes and then it sometimes produces the following message:

![travis](https://cloud.githubusercontent.com/assets/579256/15901275/da89d4e4-2dcc-11e6-9962-de968436fbd4.png)

3. No release is uploaded to GitHub as a result

### Expected behavior
I would expect every successful release to be uploaded.

**Comment by AlexisDrogoul** [comment]
Looking at the various bug reports, it is not really clear what one should do ... e.g. https://github.com/travis-ci/travis-ci/issues/2498#issuecomment-48337712

**Comment by reyman** [comment]
Hi,

When update on master are broken, the tag latest is not updated.
Actually you need to update manually the latest tag with command : 

```
git pull
git push origin :refs/tags/latest // remove the tag latest
git tag -fa latest -m "latest" //update the tag to last commit
git push 
```

**Comment by AlexisDrogoul** [comment]
OK. Thanks ! Not sure it is clearer for everyone (incl. me !), but at least we know what to do if it happens again. What does "When update on master are broken" mean, by the way ? Which updates ?

**Comment by reyman** [comment]
@AlexisDrogoul 
Sorry, my message is not clear, it's when a commit on master break the build which occur on travis, the update of the tag "latest" cannot arrive. I explain. 

The `before_deploy` step which update the tag to the last commit arrive after a success build and only for branch authorized to deploy on github :  https://github.com/gama-platform/gama/blob/master/.travis.yml#L14-L16

Github consider tag as branch.

The tag/branch "latest" is the only branch authorized to deploy, this is why you have the message "Skipping a deployment with the releases provider because this is not a tagged commit " on master log of travis.

If master failed, the mirrored branch "latest" failed, and cannot run the `travis-git-update.sh` which update the tag position to mirror the latest commit on master branch. 

If master is corrected by another commit, the "latest" branch continue to be blocked on the commit position indicated by "latest" tag because it continue to fail, and the position of tag is never updated.

If you run the command i give you, you update manually the tag position on the latest commit, and next travis build on "latest" branch reflect this update :) 

I hope this is more clear @AlexisDrogoul

**Comment by AlexisDrogoul** [comment]
Hi,

Thanks. That's more or less what I had inferred by looking at the .travis.yml and the corresponding shell file. This brings however a question: why not having "latest" being updated even if the build breaks ? This would be the best way to allow fixing a bad commit, no ?

**Comment by reyman** [comment]
@AlexisDrogoul I choose this behavior by default during coding camp, but yes, you're right. Release file are overwritted only if build success. Changing the `before_deploy` to other step recognized by travis could probably suffice : https://docs.travis-ci.com/user/customizing-the-build/ 

`before_script` could probably do the job here.

**Comment by AlexisDrogoul** [comment]
OK. I will try...

**Comment by AlexisDrogoul** [comment]
A question I have in mind regarding this is why we do not simply deploy on `master` (i.e. without any branch/tag change) ? Is there a technical limitation here ? Because it will always be the same code on `latest` and `master`...

**Comment by reyman** [comment]
Im my memory, this is a technical limitation on github. Release need to be
attached to some tag...

A question I have in mind regarding this is why we do not simply deploy on
master (i.e. without any branch/tag change) ? Is there a technical
limitation here ? Because it will always be the same code on latest and
master...

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub
<https://github.com/gama-platform/gama/issues/1876#issuecomment-225403952>,
or mute the thread
<https://github.com/notifications/unsubscribe/AAcdmR-yu5RbSjI6DzxIGKcmNvZwetBUks5qK2AygaJpZM4IxII4>
.

**Comment by hqnghi88** [comment]
Sorry for my idiot question about the deploy on master. I read in this link
https://docs.travis-ci.com/user/deployment/#Conditional-Releases-with-on%3A and look in our travis.yml . It said 2 things:

"2.branch Name of the branch. If omitted, this defaults to the app-specific branch, or master. If the branch name is not known ahead of time, you can specify all_branches: true instead of branch: ** and use other conditions to control your deployment."
"5. tags: When set to true, the application is deployed when a tag is applied to the commit. This causes the branch condition to be ignored."


Is it possible to change the
  on:
    tags: true
    all_branches: true

to
  on:
    branch: master

Cheers.

**Comment by AlexisDrogoul** [comment]
Hi Nghi,

Thanks for the update. Actually, it doesn't cost a lot to test this. I will do it ASAP and we will see... 

Alexis

**Comment by AlexisDrogoul** [comment]
Well... The result is not really convincing :)

`Skipping a deployment with the releases provider because this branch is not permitted`

**Comment by AlexisDrogoul** [comment]
Deploying using `master` now seems to work... But: Travis automatically creates an awfully named tag (see snapshot). And will probably create tags for each release now. 
![releases gama-platform gama 2016-06-13 08-44-55](https://cloud.githubusercontent.com/assets/579256/15995300/59054eb0-3143-11e6-99dc-ba58c72cc358.png)


So my question now is the following: is there a possibility

1. To make Travis work with the latest additions on `master` 
2. To let it deploy in a branch (for example one we would call `release` or `continuous release`) 

? 

How should we specify this ?  Would 
```
on:
    branch: release
``` 

be enough ?

**Comment by reyman** [comment]
Travis can be used to deploy on many remote environnent, yes. Option in
place in travis.yml are for github deployment only, and github way to
upload release only works with tag. With other type of deploiement you can
probably directly use Master.

**Comment by hqnghi88** [comment]
Hi Rey,
Thanks i am clear now. So  "git release" have its own rules. Because i have
seen the "delete and create" of "latest" tag will cause github to inform by
mails each times, so how about a upload script that upload files to a fixed
location on git? Instead of travis-git-update.sh , i think.
Cheers.
On Jun 13, 2016 12:31, "Rey" <notifications@github.com> wrote:

Travis can be used to deploy on many remote environnent, yes. Option in
place in travis.yml are for github deployment only, and github way to
upload release only works with tag. With other type of deploiement you can
probably directly use Master.
Le 12 juin 2016 5:44 PM, "hqnghi88" <notifications@github.com> a écrit :

> Sorry for my idiot question about the deploy on master. I read in this
link
>
https://docs.travis-ci.com/user/deployment/#Conditional-Releases-with-on%3A
> and look in our travis.yml . It said 2 things:
>
> "2.branch Name of the branch. If omitted, this defaults to the
> app-specific branch, or master. If the branch name is not known ahead of
> time, you can specify all_branches: true instead of branch: ** and use
> other conditions to control your deployment."
> "5. tags: When set to true, the application is deployed when a tag is
> applied to the commit. This causes the branch condition to be ignored."
>
> Is it possible to change the
> on:
> tags: true
> all_branches: true
>
> to
> on:
> branch: master
>
> Cheers.
>
> —
> You are receiving this because you were assigned.

> Reply to this email directly, view it on GitHub
> <https://github.com/gama-platform/gama/issues/1876#issuecomment-225444302
>,
> or mute the thread
> <
https://github.com/notifications/unsubscribe/AAcdmfcbrd7TelDaS-PETEUaOmyulANiks5qLClRgaJpZM4IxII4
>
> .
>

—
You are receiving this because you were assigned.

Reply to this email directly, view it on GitHub
<https://github.com/gama-platform/gama/issues/1876#issuecomment-225492292>,
or mute the thread
<https://github.com/notifications/unsubscribe/AF0pKJInRaIH0JnyrdXzYLRdZ3SxLXZmks5qLOs9gaJpZM4IxII4>
.

**Comment by AlexisDrogoul** [comment]
I think the solution proposed by Nghi can be reasonable. Actually, I find it surprising to be obliged to create/delete a tag just for the purpose of copying files into a repository. And would the alternate solution (3 messages ago) be feasible (i.e. to deploy in a branch instead of deploying to master) ?

**Comment by AlexisDrogoul** [comment]
And another question: what would happen if we simply create the `latest` tag once and dont touch it again after ? So releases could be deployed to `latest` (while using the `master` branch commits). Because one of the unwanted behavior of creating/deleting the `latest` tag is that it seems to trigger builds by Travis...

**Comment by reyman** [comment]
Sorry but i don't understand the problem here.

You can upload file to github release system only if you use tag (as you say
this is a deploy specific to github)

I delete and recreate the latest tag because this is the correct way to update
a tag. We can try to "push silently" the tag i think if you want to limit mail
messages generated by gama-bot. I have a very limited internet connection this
week, i see this later next week.

  

If you create a latest tag and never update it, release generated by travis on
this latest branch are never updated to the last commit on master.



  

  

  
Sent from [Nylas N1](https://nylas.com/n1?ref=n1), the extensible, open source
mail client.  

On juin 14 2016, at 3:46 am, Alexis Drogoul &lt;notifications@github.com&gt;
wrote:  

> And another question: what would happen if we simply create the `latest` tag
once and dont touch it again after ? So releases could be deployed to `latest`
(while using the `master` branch commits). Because one of the unwanted
behavior of creating/deleting the `latest` tag is that it seems to trigger
builds by Travis...

>

> —  
> You are receiving this because you were assigned.  
> Reply to this email directly, [view it on GitHub](https://github.com/gama-
platform/gama/issues/1876#issuecomment-225758457), or [mute the
thread](https://github.com/notifications/unsubscribe/AAcdmdTJrn_IyNBg88CuRaAXK3Gb_tcrks5qLggSgaJpZM4IxII4).![](https://github.com/notifications/beacon/AAcdmUjyKkT_fGKhnwr7_rLNrcoVsXqEks5qLggSgaJpZM4IxII4.gif)

**Comment by hqnghi88** [comment]
Dear all,
I have on my branch "travis-test" a working script that upload the zips to the tag "latest" without deleting and creating a tag. I can avoid the notification of github by mail. But i dont have much time to see how to change the status of the tag, like the comment of commit. So is it possible to push it on the master branch for further testing?

**Comment by reyman** [comment]
Hi,  i answer you when im back of the sommer school. I continue to think
this is a bad idea to remove the create / delete tag part because
http://stackoverflow.com/a/8044605/385881

Changing, remove notification are explained in the Travis doc
https://docs.travis-ci.com/user/notifications/#Email-notifications

Dear all,
I have on my branch "travis-test" a working script that upload the zips to
the tag "latest" without deleting and creating a tag. I can avoid the
notification of github by mail. But i dont have much time to see how to
change the status of the tag, like the comment of commit. So is it possible
to push it on the master branch for further testing?

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub
<https://github.com/gama-platform/gama/issues/1876#issuecomment-226452297>,
or mute the thread
<https://github.com/notifications/unsubscribe/AAcdmcGlDIYJU72rhuoHrwN89Ot-eZIpks5qMSq2gaJpZM4IxII4>
.

**Comment by AlexisDrogoul** [comment]
Hi,

There are 2 problems, maybe related to each other, maybe not. 

1) GitHub does not allow to mute certain notifications, which means each creation/deletion of the tag is notified to everyone. Of course we can filter these messages, but it's really a pain to have to do it individually (esp. since they seem to change a bit from time to time). 

2) It seems that in some configurations, the creation/deletion of the tag acts as a trigger for Travis itself, which is just plainly stupid. It happened last week, with one hundred builds or more of the rigorously same code base. 

Basically, what we need is fairly simple: that the built executables are deployed somewhere for someone to download them. If it happens to be too complicated for GitHub to manage this without creating/deleting tags or branches, then the best solution is probably to push it to the gama-platform server using plain FTP transfer. Old fashioned, but robust and not obtrusive.

**Comment by reyman** [comment]
One simple other way to remove tag notification by mail (to gama dev) is to filter directly the redirection of this mail into the gama bot gmail adress, using filter, no ?

**Comment by AlexisDrogoul** [comment]
@reyman actually, it is already off. The problem is that, individually, most of the developers are already listening to the repository, and this creates too many useless (and not really explicit) mails.

**Comment by AlexisDrogoul** [comment]
Now that we have (more or less) "silenced" travis (at least kept it quiet), another issue arises, apparently randomly: an out-of-memory error that does not seem to be linked with the changes effectively made in a commit. Is there a way to tell travis to use a little bit more memory ?

**Comment by AlexisDrogoul** [comment]
Seems to be ok now. Travis outputs things without errors and without warning the entire world that it has worked. Only problems should be reported. I close this issue, then.
